modules/ROOT/pages/migrate-client-apps.adoc

```diff
@@ -0,0 +1,154 @@
+* (optional) `url`: <URL of the client application>
++
+This property is displayed on the application page.
+
+[[migrate_with_external]]
+== Migrate Client Applications for APIs with External Client Providers
+
+When creating an application for an organization with registered external client providers, you must create the client application referencing the API that you want to consume. By doing so, you create the client application on the client provider to which the API is associated.
+
+When referencing an API that is associated to PingFederate external client provider or to the Anypoint Platform native client provider, you must include the client ID and client secret. 
+
+when creating applications referencing APIs associated with a provider like Okta or Keycloak, registered through OpenID Dynamic registration, you will not be able to define client ID and client secret credentials. This is because the OpenID Connect Dynamic Client Registration specification does not include the client ID and client secret when creating a client.
+
+To reference the API, you must retrieve the API instance ID, which is available from the API Manager UI.
+
+To migrate your client application:
+
+. Open a text editor and paste the following commands:
++
+[source,text,linenums]
+----
+curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v1/organizations/:orgId/applications?apiInstanceId=:myApiInstanceId' \
+--header 'Authorization: Bearer <omitted>' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "clientId": "myIdpClientID",
+    "clientSecret": "myIdpClientSecret",
+    "grantTypes": [
+        "client_credentials"
+    ],
+    "name": "My Application Name",
+    "redirectUri": [
+        "https://anypoint.mulesoft.com/apiplatform/:orgId/authentication/oauth2.html",
+        "https://anypoint.mulesoft.com/apiplatform/:orgId/admin/authentication/oauth2.html",
+        "https://api-notebook.anypoint.mulesoft.com/authenticate/oauth.html"
+    ]
+}'
+----
+. In the command snippet, replace the `orgId` parameter with your organization Id, `apiInstanceId` with the API instance ID, and the content of the body with the appropriate application <<properties2,properties>>. 
+. Run this command at the Command prompt.
+
+[[properties2]]
+== Properties Reference for Migrating Applications with External Provider
+
+//Tomas--can we have just one common section of properties for both migrate tasks--external provider and native provider
+In the properties of the migrate command, provide values based on the following recommendations:
+
+* `clientId`: <Your client ID>
+* `clientSecret`: <Your client secret>
+* (optional)`description`: <Informative description for the applications>
++
+This property is displayed on the application page. 
+* `grantTypes`: As no IDP is provided, leave an empty array 
+* `name`: <Name of the application to be created>
++
+This property is displayed when selecting an application before requesting contracts.
```

Comment:
```suggestion
This property is displayed when selecting an application before requesting access.
```
You request access. If it is approved, a contract is created.

Reply:
Done! Thanks!